sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -825,7 +825,7 @@ case class NTile(buckets: Expression) extends RowNumberLike with SizeBasedWindow
     zero,
     zero,
     zero,
-    (n.cast(DecimalType.IntDecimal) / buckets.cast(DecimalType.IntDecimal)).cast(IntegerType),
+    (n div buckets).cast(IntegerType),
```

Comment:
Looks like it's a partial revert of https://github.com/apache/spark/commit/ec8a1a8e8869b000a8001126ca013b753acddf6a

Reply:
yea, it's a new fix to replace https://github.com/apache/spark/commit/ec8a1a8e8869b000a8001126ca013b753acddf6a